app/src/main/java/org/piwigo/ui/main/MainActivity.java

```diff
@@ -343,31 +409,37 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         if (requestCode == SELECT_PICTURES) {
             if (resultCode == RESULT_OK) {
-                if (data.getData() != null) {
-
-                    String imageName = "";
-                    Uri targetUri = data.getData();
-                    if (data.toString().contains("content:")) {
-                        imageName = getRealPathFromURI(targetUri);
-                    } else if (data.toString().contains("file:")) {
-                        imageName = targetUri.getPath();
-                    } else {
-                        imageName = null;
-                        /* TODO add proper error handling */
-                    }
-
+                if (data.getClipData() != null) {
```

Comment:
Are you sure that we can always only used getClipData and never use "getData" (additionally)?